src/instructlab/process/process.py

```diff
@@ -0,0 +1,296 @@
+# Standard
+from datetime import datetime
+import json
+import logging
+import os
+import signal
+import subprocess
+import sys
+import time
+import uuid
+
+# Third Party
+import psutil
+
+# First Party
+from instructlab.configuration import DEFAULTS
+from instructlab.defaults import ILAB_PROCESS_MODES
+
+logger = logging.getLogger(__name__)
+
+
+def load_registry():
+    process_registry = {}
+    """Load the process registry from a file, if it exists."""
+    # we do not want a persistent registry in memory. This causes issues when in scenarios where you switch registry files (ex, in a unit test, or with multuple users)
+    # but the registry with incorrect processes still exists in memory.
+    if os.path.exists(DEFAULTS.REGISTRY_FILE):
+        with open(DEFAULTS.REGISTRY_FILE, "r") as f:
+            data = json.load(f)
+            for key, value in data.items():
+                process_registry[key] = value
+    else:
+        logger.info("No existing process registry found. Starting fresh.")
+    return process_registry
+
+
+def save_registry(process_registry):
+    """Save the current process registry to a file."""
+    with open(DEFAULTS.REGISTRY_FILE, "w") as f:
+        json.dump(dict(process_registry), f)
+
+
+class Tee:
+    def __init__(self, log_file):
+        """
+        Initialize a Tee object.
+
+        Args:
+            log_file (str): Path to the log file where the output should be written.
+        """
+        self.log_file = log_file
+        self.terminal = sys.stdout
+        self.log = log_file  # Line-buffered
+
+    def write(self, message):
+        """
+        Write the message to both the terminal and the log file.
+
+        Args:
+            message (str): The message to write.
+        """
+        self.terminal.write(message)
+        self.log.write(message)
+
+    def flush(self):
+        """
+        Ensure all data is written to the terminal and the log file.
+        """
+        self.terminal.flush()
+        self.log.flush()
+
+    def close(self):
+        """
+        Close the log file.
+        """
+        if self.log:
+            self.log.close()
+
+
+def add_process(process_mode: str, process_type: str, target, extra_imports, **kwargs):
+    """
+    Start a detached process using subprocess.Popen, logging its output.
+
+    Args:
+        task_name (str): Name of the task.
+        process_registry (dict): Shared process registry.
```

Comment:
These don't match the function args

Reply:
sorry, there might be a few of these. Lots of change here. Will go through and update.